x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -44,8 +52,17 @@ public LogicalPlanBuilder(ParserParams params) {
         super(params);
     }
 
+    private Attribute fieldTimestamp() {
+        return new UnresolvedAttribute(Source.EMPTY, params.fieldTimestamp());
+    }
+
     @Override
     public LogicalPlan visitEventQuery(EqlBaseParser.EventQueryContext ctx) {
+        return new Project(source(ctx), visitEventFilter(ctx.eventFilter()), emptyList());
```

Comment:
The projection was introduced since:
1. the results only care about the hits, not individual fields.
2. having each field extracted, due to the number of exact types, causes the doc values limit to be reached.

Before this PR the event filters did not return any attributes but now, as inputs to sequences they have to otherwise the key or timestamp field resolution for them fails.

Hence why the `visitEventQuery` now wraps the `visitEventFilter` - the latter creates the proper plan for internal use while the former wraps it up for external consumption. Note that sequence and join use the same idea - they don't output any attribute since their entire content is needed.

Reply:
@costin thx for explaining!